pandas/core/internals/blocks.py

```diff
@@ -580,8 +580,13 @@ def astype(self, dtype, copy: bool = False, errors: str = "raise"):
 
         # force the copy here
         if self.is_extension:
-            # TODO: Should we try/except this astype?
-            values = self.values.astype(dtype)
+            try:
+                values = self.values.astype(dtype)
```

Comment:
instead of this
add in this keyword to the EA astype

it's a little more code (as likely you need to refactor into an _astype method

but better i think

Reply:
I can add this, although it becomes _a lot_ more code. I was able to pull a tiny bit into a helper function, but mostly each method seems to have its own idiosyncrasies that makes it hard to abstract away details. This became quite messy so let me know if I should revert and maybe leave as a follow-up enhancement.

Reply:
hmm, Changing the signature on the base extension array is an api change. I'm not so sure if this is a good idea since this only fixes pandas EAs and third party EAs that have overridden astype will need to update their code to support this.

Reply:
It seems too large of a change for a bug fix, went ahead and reverted